File: selectionsort.py
from random import randint

def selectionSort(data):
    for i in range(len(data)):
        minIndex = i
        for j in range(i+1, len(data)):
            if data[minIndex] > data[j]:
                minIndex = j
        data[i], data[minIndex] = data[minIndex], data[i]

random_list = [randint(1,25) for i in range(10)]
print(random_list)
selectionSort(random_list)
print(random_list)

File: SelectionSort.cs
using System;
using System.Linq;

namespace SelectionSort
{
    class SelectionSort
    {
        static void Main()
        {
            Random random = new Random();

            var arr = Enumerable
                .Repeat(0, 10)
                .Select(x => random.Next(1,25))
                .ToArray();

            Console.WriteLine("Unsorted array: ");
            PrintArray(arr);
            SelectionSort(arr);
            Console.WriteLine("Sorted array: ");
            PrintArray(arr);
        }

        static void SelectionSort(int[] data)
        {
            for (int i = 0; i < data.Length - 1; i++)
            {
                int minIndex = i;
                for (int j = i + 1; j < data.Length; j++)
                {
                    if (data[j] < data[minIndex])
                    {
                        minIndex = j;
                    }
                }
                var temp = data[minIndex];
                data[minIndex] = data[i];
                data[i] = temp;
            }

        }

        static void PrintArray(int[] data)
        {
            for (int i = 0; i < data.Length; i++)
            {
                Console.WriteLine(data[i]);
            }
        }
    }
}
